sdk/compiler/damlc/lib/DA/Cli/Damlc/Command/MultiIde.hs

```diff
@@ -152,71 +178,123 @@ addNewSubIDEAndSend miState home msg = do
               , ideInHandle = inHandle
               , ideInHandleChannel = toSubIDEChan
               , ideOutHandleAsync = subIDEToCoord
+              , ideErrHandle = errHandle
               , ideProcess = subIdeProcess
-              , ideHomeDirectory = home
+              , ideHome = home
               , ideMessageIdPrefix = T.pack $ show pid
               , ideUnitId = unitId
               }
           ideData' = ideData {ideDataMain = Just ide}
 
-      putReqMethodSingleFromClient (fromClientMethodTrackerVar miState) initId LSP.SInitialize
+      -- Must happen before the initialize message is added, else it'll delete that
+      unrespondedRequests <- getUnrespondedRequestsToResend miState home
+
+      debugPrint miState "Sending init message to SubIDE"
+      putSingleFromClientMessage miState home initMsg
       putChunk inHandle $ Aeson.encode initMsg
+
       -- Dangerous calls are okay here because we're already holding the subIDEsVar lock
       -- Send the open file notifications
+      debugPrint miState "Sending open files messages to SubIDE"
       forM_ (ideDataOpenFiles ideData') $ \path -> do
         content <- TE.readFileUtf8 path
         unsafeSendSubIDE ide $ openFileMessage path content
+      
+
+      -- Resend all pending requests
+      -- No need for re-prefixing or anything like that, messages are stored with the prefixes they need
+      -- Note that we want to remove the message we're sending from this list, to not send it twice
+      let mMsgLspId = mMsg >>= fromClientRequestLspId
+          requestsToResend = filter (\req -> fromClientRequestLspId req /= mMsgLspId) unrespondedRequests
+      debugPrint miState $ "Found " <> show (length requestsToResend) <> " unresponded messages, resending."
+      debugPrint miState $ show $ (\r -> (fromClientRequestMethod r, fromClientRequestLspId r)) <$> requestsToResend
+
+      traverse_ (unsafeSendSubIDE ide) requestsToResend
+
+      debugPrint miState $ "Sending intended message to SubIDE: " <> show ((\r -> (fromClientRequestMethod r, fromClientRequestLspId r)) <$> mMsg)
       -- Send the intended message
-      unsafeSendSubIDE ide msg
+      forM_ mMsg $ unsafeSendSubIDE ide
 
       atomically $ putTMVar (subIDEsVar miState) $ Map.insert home ideData' ides
 
-runSubProc :: FilePath -> IO (Process Handle Handle ())
+disableIdeDiagnosticMessages :: SubIDEData -> FilePath -> [LSP.FromServerMessage]
+disableIdeDiagnosticMessages ideData home =
+  fullFileDiagnostic 
+    ( "GHCIDE refuses to start with the follow error:\n"
```

Comment:
typo - following error. also, is GHCIDE the correct name here?

Reply:
Hm, yeah unsure what it should be.
Maybe "IDE Environment failed to start with the following error:"?